### PR TITLE
fix APL address length check.

### PR DESCRIPTION
**Pull request by mimuret**
fix AFDLENGTH  and AFDPART length check.

### CODE REVIEW

File: msg_helpers.go [hunk 1]
```diff
@@ -783,11 +783,7 @@ func unpackDataAplPrefix(msg []byte, off int) (APLPrefix, int, error) {
 	if int(prefix) > 8*len(ip) {
 		return APLPrefix{}, len(msg), &Error{err: "APL prefix too long"}
 	}
-
 	afdlen := int(nlen & 0x7f)
-	if (int(prefix)+7)/8 != afdlen {
```
Comment (tmthrgd) [inline msg_helpers.go]: Removing this check and adding a new one below will allow `last := ip[afdlen-1]` below to panic.

Comment (mimuret) [reply msg_helpers.go]: thanks, update last octet check.

File: msg_helpers.go [hunk 2]
```diff
@@ -783,28 +783,31 @@ func unpackDataAplPrefix(msg []byte, off int) (APLPrefix, int, error) {
...
-		return APLPrefix{}, len(msg), &Error{err: "invalid APL address length"}
+	if afdlen > len(ip) {
+		return APLPrefix{}, len(msg), &Error{err: "AFDLENGTH too long"}
```
Comment (miekg) [inline msg_helpers.go]: Make this something like: 'APL length too long' ?

File: msg_helpers_test.go [hunk 3]
```diff
@@ -397,12 +397,16 @@ func TestUnpackDataAplPrefix_Errors(t *testing.T) {
...
+			[]byte{0x00, 0x01, 22, 0x05, 192, 0, 2, 0, 0},
+		},
 	}
-	for _, tt := range tests {
+	for i, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			_, _, err := unpackDataAplPrefix(tt.wire, 0)
 			if err == nil {
-				t.Fatal("expected error, got none")
+				t.Fatalf("[%d] sexpected error, got none", i)
```
Comment (miekg) [inline msg_helpers_test.go]: name should be enough here, so please revert this, so we have tiny, well contained PR

**Comment by miekg** [comment]
build is failing, can you check @mimuret ?

**Comment by mimuret** [comment]
sorry fix it.

**Pull request merged**